### PR TITLE
ncm-metaconfig: add zkrsync config file 

### PR DESCRIPTION
**Pull request by kwaegema**
config file for https://github.com/hpcugent/vsc-zk

### CODE REVIEW

File: ncm-metaconfig/src/main/metaconfig/zkrsync/main.tt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[MAIN]
+[% FOR pair IN CCM.contents.pairs -%]
+[%      pair.key %]=
+[%-      IF pair.value.is_boolean -%]
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/zkrsync/main.tt]: put the value generation in a separate tt file that has no newlines whatsoever, and then do
```
[% FOR pair IN CCM.contents.pairs -%]
[%      pair.key %]=[% INCLUDE 'metaconfig/zkrsync.value.tt' data=pari.value %]
[% END %]
```

**Comment by kwaegema** [comment]
@stdweird done

File: ncm-metaconfig/src/main/metaconfig/zkrsync/pan/schema.pan [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        error("Zkrsync client must be source OR destination");
+    } else {
+        true;
+    };
+};
+
+@{ type for configuring the zkrsync config file @}
+type zkrsync_config = {
+    # zk opts
+    'servers' : string[]
```
Comment (jrha) [inline ncm-metaconfig/src/main/metaconfig/zkrsync/pan/schema.pan]: `type_hostname[]` ?

**Comment by jrha** [comment]
Would be good to see this in 16.4, but could the validation be tightened up?

**Comment by kwaegema** [comment]
@jrha done!

**Comment by jrha** [comment]
Cool, have you tested to make sure it still works with the tighter validation?

**Comment by kwaegema** [comment]
@jrha yes, the unit test covers this :smile:

**Pull request merged**